OMGVA PoS/ClientApp/src/pages/UpdateBusinessPage.tsx

```diff
@@ -0,0 +1,68 @@
+﻿import React, { useState, useEffect } from 'react';
+import { useParams, useNavigate, useLocation } from 'react-router-dom'
+import { Business, updateBusiness } from '../services/businessService';
+import { getTokenBusinessId, getTokenRole } from '../utils/tokenUtils';
+import '../index.css';
+import BusinessDataForm from '../components/Form/BusinessDataform';
+
+interface UpdateBusinessPageProps {
+    token: string | null;
+}
+
+const UpdateBusinessPage: React.FC<UpdateBusinessPageProps> = ({ token: authToken }) => {
+    const [error, setError] = useState<string | null>(null);
+    const [business, setBusiness] = useState<Business>();
+    const { state } = useLocation();
+    const { id } = useParams();
+    const navigate = useNavigate();
+
+    const handleSubmission = async (businessInfo: Business) => {
+        try {
+            if (id) {
+                const error = await updateBusiness(authToken, id, businessInfo);
+                if (error) {
+                    setError("An error occurred while updating business: " + error);
+                    return;
+                }
+                navigate('/business/' + id);
+            }
+            else {
+                setError("The id of the business is missing");
```

Comment:
Same right here with errors - our client won't put the business id mechanically - we should do it for him. Maybe different message here?

Reply:
In reality this error should absolutely never happen as you should not be able to navigate to this page without providing the id. Also, the id is in the url, so it is technically a possibility. Honestly, I am not sure how to change this error.

Reply:
Let's put "Bad request" and that's it?

Reply:
Maybe "Could not identify the business". It is not really a request.

Reply:
Changed it.